envoy-control-services/src/test/kotlin/pl/allegro/tech/servicemesh/envoycontrol/services/MultiClusterStateTest.kt

```diff
@@ -0,0 +1,20 @@
+package pl.allegro.tech.servicemesh.envoycontrol.services
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+
+internal class MultiClusterStateTest {
+
+    @Test
+    fun `MultiCLusterStates should implement equality`() {
+        // given
+        val element = ServiceInstance("1", address = "0.0.0.0", port = 1, tags = setOf("a"))
+        val serviceInstances = ServiceInstances("a", setOf(element))
+        val l = listOf(ClusterState(ServicesState(mapOf("a" to serviceInstances)), Locality.REMOTE, "dc1"))
```

Comment:
IMO equality test should test totally different objects, not objects sharing the same fields. I mean: `element`, `serviceInstances` and `l` should not be shared between these two objects.

The easiest way do to it is to create object representation in json and parse it two times.

Currently this test is very limited, because it will not catch broken equality in for example `ServiceInstance`